File: Aula 7/java/desafio.java
public class Insumo {

    // Passo 1: Estrutura do Insumo

    private String nome;
    private int quantidade;
    private int mesValidade;

    public Insumo(String nome, int quantidade, int mesValidade) {

        this.nome = nome;
        this.quantidade = quantidade;

        // valor padrão
        this.mesValidade = 1;

        // usa o método validador
        setMesValidade(mesValidade);
    }

    // Passo 2: Filtro de Sanidade

    public void setMesValidade(int mes) {

        if (mes >= 1 && mes <= 12) {

            this.mesValidade = mes;

            System.out.println(
                "Mês de validade atualizado para " + mes
            );

        } else {

            System.out.println(
                "ERRO: mês inválido! Digite um valor entre 1 e 12."
            );
        }
    }

    // Passo 3: Trava de Segurança Final

    public boolean estaValido(int mesAtual) {

        return this.mesValidade >= mesAtual;
    }

    // Método extra para mostrar dados

    public void mostrarDados() {

        System.out.println("Insumo: " + this.nome);
        System.out.println("Quantidade: " + this.quantidade);
        System.out.println("Mês de validade: " + this.mesValidade);
    }
}

File: Aula 7/java/desafio.py
class Insumo:

    # Passo 1: Estrutura do Insumo
    def __init__(self, nome, quantidade, mes_validade):

        self.__nome = nome
        self.__quantidade = quantidade
        self.__mes_validade = 1

        # Usa o método validador
        self.set_mes_validade(mes_validade)

    # Passo 2: Filtro de Sanidade
    def set_mes_validade(self, mes):

        if 1 <= mes <= 12:
            self.__mes_validade = mes
            print(f"Mês de validade atualizado para {mes}")

        else:
            print("ERRO: mês inválido! Digite um valor entre 1 e 12.")

    # Passo 3: Trava de Segurança Final
    def esta_valido(self, mes_atual):

        return self.__mes_validade >= mes_atual

    # Método extra para mostrar dados
    def mostrar_dados(self):

        print(f"Insumo: {self.__nome}")
        print(f"Quantidade: {self.__quantidade}")
        print(f"Mês de validade: {self.__mes_validade}")


# TESTES

amendoim = Insumo("Amendoim", 50, 10)

print("\n--- Dados do Produto ---")
amendoim.mostrar_dados()

print("\n--- Testando Validade ---")
print(amendoim.esta_valido(8))   # True
print(amendoim.esta_valido(11))  # False

print("\n--- Testando erro ---")
amendoim.set_mes_validade(13)